packages/twenty-front/src/modules/action-menu/actions/record-actions/components/RecordActionMenuEntriesSetter.tsx

```diff
@@ -32,30 +32,36 @@ const ActionEffects = ({
     objectId: objectMetadataItemId,
   });
 
-  const contextStoreNumberOfSelectedRecords = useRecoilComponentValueV2(
-    contextStoreNumberOfSelectedRecordsComponentState,
+  const contextStoreTargetedRecordsRule = useRecoilComponentValueV2(
+    contextStoreTargetedRecordsRuleComponentState,
   );
 
   const isWorkflowEnabled = useIsFeatureEnabled('IS_WORKFLOW_ENABLED');
 
   return (
     <>
-      {contextStoreNumberOfSelectedRecords === 0 && (
-        <NoSelectionActionMenuEntrySetterEffect
-          objectMetadataItem={objectMetadataItem}
-        />
-      )}
-      {contextStoreNumberOfSelectedRecords === 1 && (
-        <SingleRecordActionMenuEntrySetterEffect
-          objectMetadataItem={objectMetadataItem}
-        />
-      )}
-      {contextStoreNumberOfSelectedRecords === 1 && isWorkflowEnabled && (
-        <WorkflowRunRecordActionMenuEntrySetterEffect
-          objectMetadataItem={objectMetadataItem}
-        />
-      )}
-      {contextStoreNumberOfSelectedRecords > 1 && (
+      {contextStoreTargetedRecordsRule.mode === 'selection' &&
+        contextStoreTargetedRecordsRule.selectedRecordIds.length === 0 && (
+          <NoSelectionActionMenuEntrySetterEffect
+            objectMetadataItem={objectMetadataItem}
+          />
+        )}
+      {contextStoreTargetedRecordsRule.mode === 'selection' &&
+        contextStoreTargetedRecordsRule.selectedRecordIds.length === 1 && (
+          <SingleRecordActionMenuEntrySetter
+            objectMetadataItem={objectMetadataItem}
+          />
+        )}
+      {contextStoreTargetedRecordsRule.mode === 'selection' &&
+        contextStoreTargetedRecordsRule.selectedRecordIds.length === 1 &&
+        isWorkflowEnabled && (
+          <WorkflowRunRecordActionMenuEntrySetterEffect
+            objectMetadataItem={objectMetadataItem}
+          />
```

Comment:
why does this not live into a SingleRecordActionMenuEntrySetterEffect? Since we have a NoSelectionActionMenuEntrySetterEffect

Reply:
Because `WorkflowRunRecordActionMenuEntrySetterEffect` sets the workflow actions when a record is selected. Thus, it doesn't belong in `NoSelectionActionMenuEntrySetterEffect`.